File: src/me/Qball/Wild/Utils/WorldInfo.java
package me.Qball.Wild.Utils;

import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;

import me.Qball.Wild.Wild;

public class WorldInfo {
	Wild wild = Wild.getInstance();
	public String getWorldName(Player p){
		ConfigurationSection sec = wild.getConfig().getConfigurationSection("Worlds");
		for(String key : sec.getKeys(false))
		{
			if(key.equals(p.getWorld().getName()))
			{
				return p.getWorld().getName();
			}
		}
		return "";
}
	public int getMinX(String world)
	{
		return wild.getConfig().getInt("Worlds."+ world + ".MinX");
	}
	public int getMaxX(String world)
	{
		return wild.getConfig().getInt("Worlds."+ world + ".MaxX");
	}
	public int getMinZ(String world)
	{
		return wild.getConfig().getInt("Worlds."+ world + ".MinZ");
	}
	public int getMaxZ(String world)
	{
		return wild.getConfig().getInt("Worlds."+ world + ".MaxZ");
	}
	public void setWorldName(String world)
	{
		wild.getConfig().createSection("Worlds."+world);
		wild.saveConfig();
	}
	public void setMinX(String world, int min)
	{
		wild.getConfig().set("Worlds."+world+".MinX", min);
		wild.saveConfig();
	}
	public void setMaxX(String world, int max)
	{
		wild.getConfig().set("Worlds."+world+".MaxX", max);
		wild.saveConfig();
	}
	public void setMinZ(String world, int min)
	{
		wild.getConfig().set("Worlds."+world+".MinZ", min);
		wild.saveConfig();
	}
	public void setMaxZ(String world, int max)
	{
		wild.getConfig().set("Worlds."+world+".MaxZ", max);
		wild.saveConfig();
	}
}

File: src/main/kotlin/me/qball/wild/util/WorldInfo.kt
package me.qball.wild.util

import me.qball.wild.Wild
import org.bukkit.entity.Player

class WorldInfo(plugin: Wild) {
    private val plugin = plugin

    fun getWorldName(p: Player) : String {
        val worlds = plugin.config.getConfigurationSection("Worlds")
        worlds.getKeys(false).forEach {
            if(it.equals(p.world.name)) {
                return it
            }
        }
        return ""
    }

    fun getMinX(w: String) : Int {
        return plugin.config.getInt("Worlds.$w.MinX")
    }

    fun getMaxX(w: String) : Int {
        return plugin.config.getInt("Worlds.$w.MaxX")
    }

    fun getMinZ(w: String) : Int {
        return plugin.config.getInt("Worlds.$w.MinZ")
    }

    fun getMaxZ(w: String) : Int {
        return plugin.config.getInt("Worlds.$w.MaxZ")
    }

    fun setWorldName(world: String) {
        plugin.config.createSection("Worlds.$world")
        plugin.saveConfig()
    }

    fun setMinX(world: String, min: Int) {
        plugin.config.set("Worlds.$world.MinX", min)
        plugin.saveConfig()
    }

    fun setMaxX(world: String, max: Int) {
        plugin.config.set("Worlds.$world.MaxX", max)
        plugin.saveConfig()
    }

    fun setMinZ(world: String, min: Int) {
        plugin.config.set("Worlds.$world.MinZ", min)
        plugin.saveConfig()
    }

    fun setMaxZ(world: String, max: Int) {
        plugin.config.set("Worlds.$world.MaxZ", max)
        plugin.saveConfig()
    }
}


